homeassistant/components/media_player/cast.py

```diff
@@ -216,13 +213,15 @@ def _async_create_cast_device(hass: HomeAssistantType,
     if not isinstance(config, list):
         config = [config]
 
-    await asyncio.wait([
+    done, pending = await asyncio.wait([
         _async_setup_platform(hass, cfg, async_add_entities, None)
         for cfg in config])
+    if pending or any([not task.result() for task in done]):
+        raise PlatformNotReady
```

Comment:
What happens if the setup for two chromecast succeeded but failed for a third chromecast? Then I think this code would re-connect all three of them again. 

If the devices have no UUID (don't know if this can be the case with config entries), this could end up creating new entities for the two connectable chromecasts every 10 seconds.

Reply:
We will get the message such as
`Discovered previous chromecast ...`